homeassistant/components/amcrest/__init__.py

```diff
@@ -105,7 +104,7 @@ def setup(hass, config):
             # pylint: disable=pointless-statement
             camera.current_time
 
-        except (ConnectError, ConnectTimeout, HTTPError) as ex:
+        except (CommError, LoginError) as ex:
```

Comment:
I suggest using `AmcrestError` instead to make sure you catch all kinds of exceptions (also future ones).

Reply:
Yes, that of course would work, too. But isn't that like using `except Exception` instead of `except (ValueError, TypeError)`? I.e., I thought it's more "Pythonic" to list the expected exceptions instead of using a "catch all" exception.

Reply:
It's fine to catch your own base class, that will not hide bugs by matching e.g. an unexpected `TypeError`.

If you want to handle them differently, you will of course need to name each one.

Reply:
Okey dokey. Will do.